BepInExResoniteShim.cs

```diff
@@ -147,4 +147,19 @@ public static IEnumerable<CodeInstruction> Transpiler(IEnumerable<CodeInstructio
             return null;
         }
     }
+}
+
+public static class ResoniteHooks
+{
+    public static event Action<Engine>? OnEngineInit;
+
+    [HarmonyPatch(typeof(Engine), "SetReady")]
+    public class EngineInitPatch
+    {
+        public static void Postfix(Engine __instance)
+        {
+            BepInExResoniteShim.Logger.LogInfo("Engine initialization finished, firing OnEngineInit event");
```

Comment:
this should only fire if anything is subscribed to the OnEngineReady Event

Reply:
meaning the log msg should not show if no plugins have registered callbacks with the event